code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -490,6 +491,29 @@
 				if(G.summoner)
 					G.summoner << "<span class='danger'><B>[AM] has crossed your surveillance trap at [get_area(snare_loc)].</span></B>"
 
+/mob/living/simple_animal/hostile/guardian/ranged/Shoot(atom/targeted_atom)
+	var/obj/item/projectile/guardian/shard = ..()
+	if(!shard)
+		return
+	shard.color = guardiancolor
+
+/proc/techcolor2hex(var/colour)
```

Comment:
Unneeded proc pretty sure(Check through,i'm not 100% sure), cause color can also be a word and not necessarily an hex code